src/Test/Utilities/Portable/Metadata/ILValidation.cs

```diff
@@ -3,16 +3,200 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.IO;
 using System.Linq;
+using System.Reflection;
 using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Security.Cryptography;
 using System.Text;
+using System.Threading;
 using System.Xml;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis;
 using Microsoft.Metadata.Tools;
 
 namespace Roslyn.Test.Utilities
 {
     public static class ILValidation
     {
+        /// <summary>
+        /// Validates that the given stream is marked as signed, the signature matches
+        /// the public key, and the header checksum is correct.
+        /// </summary>
+        public static bool IsStreamFullSigned(Stream moduleContents)
+        {
+            var savedPosition = moduleContents.Position;
+
+            try
+            {
+                moduleContents.Position = 0;
+
+                using (var metadata = ModuleMetadata.CreateFromStream(moduleContents, leaveOpen: true))
+                {
+                    var metadataReader = metadata.MetadataReader;
+                    var peReader = metadata.Module.PEReaderOpt;
+                    var peHeaders = peReader.PEHeaders;
+                    var flags = peHeaders.CorHeader.Flags;
+
+                    bool is32bit = peHeaders.PEHeader.Magic == PEMagic.PE32;
```

Comment:
Consider moving `metadataReader`, `peHeaders`,   `is32bit`, `SectionHeaderSize`, `signature` (and its inputs), `signatureBlob` closer to usage too